### PR TITLE
lib3mf: 2.0.0 -> 2.1.1

### LINKED ISSUE #116888: Vulnerability roundup 100: lib3mf-2.0.0: 1 advisory [8.1]
[search](https://search.nix.gsc.io/?q=lib3mf&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=lib3mf+in%3Apath&type=Code)

* [ ] [CVE-2021-21772](https://nvd.nist.gov/vuln/detail/CVE-2021-21772) CVSSv3=8.1 (nixos-unstable)

Scanned versions: nixos-unstable: 1f77a4c8c74.

Cc @gebner

### PR DESCRIPTION
**Pull request by gebner**
###### Motivation for this change

Fixes #116888 

openscad's 3MF support has been broken since #101105, since lib3mf 2.0 has a different API.  I believe the easiest way to fix this is to wait for the next openscad release (with the new cmake build system).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116952` at 04e1a9d2 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build (new failures):</summary>
  <ul>
    <li><a href="https://gist.github.com/8dea7cd27755938dbb478b5208f0664b">lib3mf</a></li></li>
    <li><a href="https://gist.github.com/02b5d9664ce67e933dcc9cbdd357f825">openscad</a></li></li>
  </ul>
</details>

File: pkgs/development/libraries/lib3mf/default.nix [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-  version = "2.0.0";
+  version = "2.1.1";
 
   src = fetchFromGitHub {
     owner = "3MFConsortium";
     repo = pname;
     rev = "v${version}";
-    sha256 = "0w4d9zvl95g1x3r5nyd6cr27g6fwhhwaivh8a5r1xs5l6if21x19";
+    sha256 = "0qah4dgird4z1l4lhcfijr5fwnclhfxk5br7fwmqcwkws82y8nia";
+    fetchSubmodules = true;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lib3mf/default.nix:12]: I think we should unvendor libressl and googletest.

**Comment by SuperSandro2000** [comment]
Also I noticed while building that we should split out include into a dev output

---

```
lib3mf: Ask if output path include (4.0M) could be split with outputs = [ ... "dev" ];
```

File: pkgs/development/libraries/lib3mf/default.nix (hunk 1 above)
Comment (gebner) [reply pkgs/development/libraries/lib3mf/default.nix]: Fortunately upstream has added cmake flags to use the system dependencies now, so this was easy.

File: pkgs/development/libraries/lib3mf/default.nix [hunk 2]
```diff
@@ -1,26 +1,41 @@
-{ lib, stdenv, fetchFromGitHub, cmake, ninja, libuuid, libossp_uuid, gtest }:
+{ lib, stdenv, fetchFromGitHub, cmake, ninja, automaticcomponenttoolkit,
+  pkg-config, libzip, gtest, openssl, libuuid, libossp_uuid }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/lib3mf/default.nix:2]:
```suggestion
{ lib, stdenv, fetchFromGitHub, cmake, ninja, automaticcomponenttoolkit
, pkg-config, libzip, gtest, openssl, libuuid, libossp_uuid }:
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -11936,6 +11936,8 @@ in
 
   astyle = callPackage ../development/tools/misc/astyle { };
 
+  automaticcomponenttoolkit = callPackage ../development/tools/misc/automaticcomponenttoolkit {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:11939]:
```suggestion
  automaticcomponenttoolkit = callPackage ../development/tools/misc/automaticcomponenttoolkit { };
```

**Comment by mweinelt** [comment]
Fixes: CVE-2021-21772

> A use-after-free vulnerability exists in the NMR::COpcPackageReader::releaseZIP() functionality of 3MF Consortium lib3mf 2.0.0. A specially crafted 3MF file can lead to code execution. An attacker can provide a malicious file to trigger this vulnerability.

**Pull request merged**